### PR TITLE
Fixes problem with inconsistent names

### PR DESCRIPTION
**Pull request by nikolase90**
My idea behind these changes: 
- `m` should always represent the total number of features. It is a single number. 
- `n_features` could be a sequence of numbers. Every item of `n_features` should be greater or equal to zero, and less or equal to `m`. 

Note that I've also changed cases where we called it `nfeatures` instead of `n_features`. 

In a different PR I will change the name of `explainer$n_features` to `explainer$m`. I don't want to do this in this PR because that will also change the tests. 

Fixes https://github.com/NorskRegnesentral/shapr/issues/136.

### CODE REVIEW

File: R/features.R [hunk 1]
```diff
@@ -16,9 +16,9 @@
 #' \item{features}{List. Each item of the list is an integer vector where \code{features[[i]]}
 #' represents the indices of the features included in combination \code{i}. Note that all the items
 #' are sorted such that \code{features[[i]] == sort(features[[i]])} is always true.}
-#' \item{nfeatures}{Positive integer. \code{nfeatures[i]} equals the number of features in combination
-#' \code{i}, i.e. \code{nfeatures[i] = length(features[[i]])}.}.
-#' \item{N}{Positive integer. The number of unique ways to sample \code{nfeatures[i]} features
+#' \item{n_features}{Positive integer. \code{n_features[i]} equals the number of features in combination
```
Comment (martinju) [inline R/features.R]:
```suggestion
#' \item{n_features}{Vector of positive integers. \code{n_features[i]} equals the number of features in combination
```

Comment (martinju) [reply R/features.R]: Right?

Comment (nikolase90) [reply R/features.R]: Yep!

File: R/shapley.R [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #' Calculate Shapley weight
 #'
 #' @param m Positive integer. Total number of features.
-#' @param s Positive integer. Represents the number of features you want to sample from a feature
+#' @param n_features Positive integer. Represents the number of features you want to sample from a feature
 #' space consisting of \code{m} unique features. Note that \code{ 0 < = s  <= m}.
```
Comment (martinju) [inline R/shapley.R]:
```suggestion
#' space consisting of \code{m} unique features. Note that \code{ 0 < = n_features <= m}.
```

File: R/shapley.R [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 #' @param N Positive integer. The number of unique combinations when sampling \code{s} features,
```
Comment (martinju) [inline R/shapley.R]:
```suggestion
#' @param N Positive integer. The number of unique combinations when sampling \code{n_features} features,
```

File: src/impute_data.cpp [hunk 4]
```diff
@@ -13,13 +13,13 @@ using namespace Rcpp;
 //'
 //' @param xtest Numeric matrix. Represents a single test observation.
 //'
-//' @param S Integer matrix of dimension \code{ncomb x nfeatures}, where \code{ncomb} equals
-//' the total number of sampled/non-sampled feature combinations and \code{nfeatures} equals
+//' @param S Integer matrix of dimension \code{n_combinations x m}, where \code{n_combinations} equals
+//' the total number of sampled/non-sampled feature combinations and \code{m} equals
 //' the total number of unique features. Note that \code{nfeature = ncol(xtrain)}. See details
```
Comment (martinju) [inline src/impute_data.cpp]:
```suggestion
//' the total number of unique features. Note that \code{m = ncol(xtrain)}. See details
```

**Comment by nikolase90** [comment]
Awesome! Pushed one additional fix with 251cd2b. Also searched through, and couldn't find any additional cases (but I guess there are some).

**Comment by nikolase90** [comment]
Likewise!

**Pull request merged**